### PR TITLE
Reenable rating

### PR DESCRIPTION
**Pull request by kkoehn**
Reenables ratings of tasks on the show view 
also displays current average rating on index (incl filter)

todo: 
- [x] check filter
  - [x] weird stuff is happening with pagination
  - [x] reset pagination, if any filter changes
- [x] testing
- [x] specs

### CODE REVIEW

File: app/assets/javascripts/tasks_ratings.coffee [hunk 1]
```diff
@@ -0,0 +1,74 @@
+ready =->
+  initializeRatings()
+
+$(document).on('turbolinks:load', ready)
+
+initializeRatings = ->
+  USERRATING = 0
+
+#  $('.popup-rating').hover (->
```
Comment (kkoehn) [inline app/assets/javascripts/tasks_ratings.coffee:9]: this was in the old code - do you think it serves any purpose?
it seems to work fine without

File: app/assets/javascripts/tasks_ratings.coffee [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    $(lower).removeClass("fa-star-o").addClass("fa-star")
+    upper = $('.rating span').filter (->
+      $(this).attr("data-rating") > rating
+    )
+    $(upper).removeClass("fa-star").addClass("fa-star-o")
+  )
+
+  $('.rating span').on 'click', ->
+    rating = this.getAttribute("data-rating")
+#    $(this).removeClass("fa-star-o").addClass("fa-star")
```
Comment (kkoehn) [inline app/assets/javascripts/tasks_ratings.coffee:36]: see other comment

File: app/assets/javascripts/tasks_ratings.coffee [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#    USERRATING = $('.rating span.fa-star').last().attr("data-rating")
+#  ), ->
+#    lower = $('.rating span').filter (->
+#      $(this).attr("data-rating") <= USERRATING
+#    )
+#    $(lower).removeClass("fa-star-o").addClass("fa-star")
+#    upper = $('.rating span').filter (->
+#      $(this).attr("data-rating") > USERRATING
+#    )
+#    $(upper).removeClass("fa-star").addClass("fa-star-o")
```
Comment (kkoehn) [inline app/assets/javascripts/tasks_ratings.coffee:19]: this was in the old code - do you think it serves any purpose?
it seems to work fine without

Comment (MrSerth) [reply app/assets/javascripts/tasks_ratings.coffee:19]: No, not really. It seems duplicated compared to the other logic below, and I would also opt to remove it. If you do so, please also remove the `USERRATING` variable in this file -- it's not used as well, is it?

File: app/views/tasks/index.html.slim [hunk 4]
```diff
@@ -42,12 +43,15 @@
           p
             b = t('tasks.index.search.order')
           .btn-group[role="group"]
-            = sort_link(@search, :created_at)
-        /.stars.section
+            div
+              = sort_link(@search, :created_at)
+            div
+              = sort_link(@search, :average_rating)
```
Comment (MrSerth) [inline app/views/tasks/index.html.slim:49]: At least in my tests performed locally, this sorting is not working as expected. Would you mind having a look here again, please?

File: app/views/tasks/show.html.slim [hunk 5]
```diff
@@ -29,7 +29,7 @@
               - else
                 - [*1..5].each do |i|
                   span.fa.fa-star-o data-rating=i
-    = render 'remove_state_tag', exercise: @task
+    /= render 'remove_state_tag', exercise: @task
```
Comment (MrSerth) [inline app/views/tasks/show.html.slim:32]: What exactly was this `remove_state_tag` for? I cannot find it in the code any longer...

File: config/locales/en.yml [hunk 6]
```diff
@@ -1057,7 +1058,7 @@ en:
         order: "Order By"
         rating: "Rating"
         created_at: "Created At"
-        stars: "Stars"
+        min_stars: "Minimum amount of stars"
```
Comment (MrSerth) [inline config/locales/en.yml:1061]: I think stars are countable:
```suggestion
        min_stars: "Minimum number of stars"
```

**Review by MrSerth — APPROVED** [review]
Great, the rating (and also the filter including the sorting) is now working again as expected!

**Pull request merged**